sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/RestfulClient.java

```diff
@@ -0,0 +1,187 @@
+import java.util.Collection;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManagerFactory;
+
+public class RestfulClient {
+
+  private static final Logger logger = Logger.getLogger(RestfulClient.class.getName());
+
+  private static final int TIMEOUT_MILLIS = 2000;
+
+  String fetchString(
+      String httpMethod,
+      String urlStr,
+      Map<String, String> requestPropertyMap,
+      String keyStorePath) {
+    final HttpURLConnection connection;
+
+    try {
+      if (Strings.isNullOrEmpty(keyStorePath)) {
+        connection = (HttpURLConnection) new URL(urlStr).openConnection();
+      } else {
+        connection = (HttpsURLConnection) new URL(urlStr).openConnection();
+        ((HttpsURLConnection) connection).setSSLSocketFactory(buildSslSocketFactory(keyStorePath));
+      }
+
+      connection.setRequestMethod(httpMethod);
+      connection.setConnectTimeout(TIMEOUT_MILLIS);
+      connection.setReadTimeout(TIMEOUT_MILLIS);
+
+      if (requestPropertyMap != null) {
+        for (Map.Entry<String, String> requestProperty : requestPropertyMap.entrySet()) {
+          connection.setRequestProperty(requestProperty.getKey(), requestProperty.getValue());
+        }
+      }
+
+      int responseCode = connection.getResponseCode();
+      if (responseCode != 200) {
+        logger.log(
+            Level.WARNING,
+            String.format(
+                "Error reponse from %s code (%s) text %s",
+                urlStr, responseCode, readResponseString(connection)));
+        return "";
+      }
+
+      return readResponseString(connection).trim();
+
+    } catch (IOException e) {
+      logger.log(Level.WARNING, String.format("Restful Client fetch string failed: %s", e));
+    }
+
+    return "";
+  }
+
+  private static String readResponseString(HttpURLConnection connection) {
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    try (InputStream is = connection.getInputStream()) {
+      ByteStreams.copy(is, os);
+    } catch (IOException e) {
+      // Only best effort read if we can.
+    }
+    try (InputStream is = connection.getErrorStream()) {
+      if (is != null) {
+        ByteStreams.copy(is, os);
+      }
+    } catch (IOException e) {
+      // Only best effort read if we can.
+    }
+    try {
+      return os.toString(StandardCharsets.UTF_8.name());
+    } catch (UnsupportedEncodingException e) {
+      throw new IllegalStateException("UTF-8 not supported can't happen.", e);
+    }
+  }
+
+  private static SSLSocketFactory buildSslSocketFactory(String keystorePath) {
+    try {
+      KeyStore keyStore = getK8sKeystore(keystorePath);
+
+      String tmfAlgorithm = TrustManagerFactory.getDefaultAlgorithm();
+      TrustManagerFactory tmf = TrustManagerFactory.getInstance(tmfAlgorithm);
+      tmf.init(keyStore);
```

Comment:
Let's avoid doing anything if `keyStore` is null due to an error. So I guess in the caller,

```java
KeyStore keyStore = getK8sKeystore(keystorePath);
if (keyStore != null) { ((HttpsURLConnection))connection.setSslSocketFactory(buildSslSocketFactory(keyStore));
```

Reply:
fix in next commit